src/main/java/at/hannibal2/skyhanni/utils/NumberUtil.kt

```diff
@@ -212,4 +214,18 @@ object NumberUtil {
     fun Number.fractionOf(maxValue: Number) = maxValue.toDouble().takeIf { it != 0.0 }?.let { max ->
         this.toDouble() / max
     }?.coerceIn(0.0, 1.0) ?: 1.0
+
+    fun interpolate(now: Float, last: Float, lastUpdate: Long): Float {
+        var interp = now
+        if (last >= 0 && last != now) {
+            var factor: Float = (SimpleTimeMark.now().toMillis() - lastUpdate) / 1000f
+            factor = clampZeroOne(factor)
+            interp = last + (now - last) * factor
+        }
+        return interp
+    }
+
+    private fun clampZeroOne(f: Float): Float {
```

Comment:
look at the logic of fractionOf above. either replace this function with coerceIn(0,1), or use this function in fractionOf instead. right now it looks like code duplication